pkg/sosreport/index.html

```diff
@@ -28,20 +28,124 @@
+        <div class="pf-c-backdrop">
+            <div class="pf-l-bullseye">
+                <div class="pf-c-modal-box pf-m-md pf-m-align-top" role="dialog" aria-modal="true">
+                    <button class="pf-c-button pf-m-plain" type="button" aria-label="Close" id="sos-options-close-button">
+                        <i class="fa fa-times" aria-hidden="true"></i>
+                    </button>
+                    <header class="pf-c-modal-box__header">
+                        <h1 class="pf-c-modal-box__title" translate="yes">Set SoS Report Options</h4>
+                        <small>All settings are optional, if in doubt leave default values and click 'Run Report'</small>
+                    </header>
+                    <div class="pf-c-modal-box__body">
+                        <div id="sos-settings-body">
+                            <form class="ct-form" id="sos-opts-form">
+                                <label class="control-label" for="sos-label">Report Label</label>
+                                <input id="sos-label" class="form-control" type="text" title="Append a label/name to the final report archive name. No spaces.">
+
+                                <label class="control-label" for="sos-case-id">Case Number</label>
+                                <input id="sos-case-id" class="form-control" title="If you have a support case open with a vendor, enter that case id here">
+
+                                <label class="control-label" for="sos-log-size">Log Size Limit</label>
+                                <input id="sos-log-size" class="form-control" title="Maximum size in MB to collect for each log file or command output (default 25)">
+
+                                <label class="control-label" for="sos-threads">Threads</label>
+                                <select id="sos-threads" class="form-control" title="Number of plugins to run concurrently">
+                                    <!-- 4 is the default, and more than that shows significant diminishing returns
+                                         so only provide option to reduce concurrent plugin count -->
+                                    <option value=1>1</option>
+                                    <option value=2>2</option>
+                                    <option value=3>3</option>
+                                    <option value=4 selected>4</option>
+                                </select>
+
+                                <label class="control-label" for="sos-skip-plugins">Skip These Plugins</label>
+                                <input id="sos-skip-plugins" class="form-control" title="List of plugin names to skip, comma delimited">
+
+                                <label class="control-label" for="sos-only-plugins">Only Run These Plugins</label>
+                                <input id="sos-only-plugins" class="form-control" title="List of plugin names to ONLY run, comma delimited">
```

Comment:
These two are almost unusable. Can we get list of all plugins and then show it in the UI?
Possibly [multi select](https://www.patternfly.org/v4/components/select#multiple) with pre-selected plugins that are auto enabled.

Reply:
Actually, only one at a time makes sense, doesn't it?
If you pick some "to only run" then "skip these" doesn't make a lot of sense.
So only having one field with the auto enabled ones and making it easy to deselect all to make 'only run these' easy to do.

Reply:
So right now there are over 300 plugins in sos. Even on super sparse distributions you're looking at ~50 default enabled plugins, so I think any form of select here will be rather clunky/unwieldy. The form inputs match the CLI option syntax - a comma delimited string.

At first glance yes, only having "skip" or "only" available makes sense, however there are use cases where both can be used at the same time. For example, imagine that `/etc/sos/sos.conf` specifies to always run the `foo` profile, which sets it's own defined set of plugins. However, the user knows that they want to modify the plugin list slightly. If they use `skip` they can then disable plugins that were otherwise enabled by the `foo` profile, and similarly if they use the `only` field they can add additional plugins on top of what the profile enables.

Reply:
I'm not convinced about this either. This really isn't something where a human writes down five plugins with precise names without error. If you need such fine-grained control, IMHO you are better off with actually *using* the command line -- this UI isn't meant to be (basically) a GUI widget for typing a command line.

IMHO a good middle ground would be to only list available profiles, such as `security`, 'system`, `hardware`. There's 22 of them, which is reasonable. WDYT?

Reply:
A profile dropdown would be handy, and could be easily populated by scraping `sos report --list-profiles` (if that's acceptable? Or would this need to be a curated select?).

My concern with not having the skip option exposed at all is more historical from generating a report from CLI. There's a long history of certain plugins (e.g. 'logs') exploding archive size due to some other issue, or when combined with other options like `--all-logs`. E.G. the user is thinking "I need all logs for X, Y, and Z plugins, but I want to skip the logs plugin because I don't need to entire system journal for this system". We've made several safeguards against hung plugins in the past few releases, but combining `--all-logs` with "give me the system journal" is still something that can surprise users.

That being said, if this should just be targeting the very simple "give me a default report" use case, and anything else should defer to using the CLI directly, then that makes sense and I could drop off a number of the other options in this PR too. Just have the basics - label, case number, default log size, and the `--clean` toggle.

Does that sound better than providing any of the plugin/profile/preset controls?

Reply:
I think showing `sos report --list-profiles` would be acceptable (is there some summary which the page could parse and show?) Note that by default programs are being called with the selected browser locale, so if list-profiles shows i18n'ed explanation strings, they'd even be shown in the UI. It is not maintainable/desirable to have such a "profile name" → "human description" mapping in the cockpit plugin. But even with just the English short names these should already be helpful.

As to the desired use cases, you are a much better person to tell than us Cockpit developers. We are mostly trying to give some guidance/guard rails as to usability and such, so please don't interpret it as "we don't want this" -- we do! FWIW, if you think it's useful to discuss this with high bandwidth, let's schedule a call.

In its current form I certainly find the dialog overwhelming (it would certainly be the biggest dialog in Cockpit by far), so it does make sense to think about some balance between flexibility and comprehension. Clearly the current page is too simple -- you made this PR for a reason, and I also saw a similar request in bugzilla in the past.

My personal gut feeling is that 22 pre-defined profiles with reasonably understandable names are about the right granularity, whereas free-form "computer text" input for 300+ plugins is too much. You make a convincing case for an option to include (or not) the full journal (even though I think that's by far too much -- it's usually many gigabytes going back a year or even more!), but as you say that could have a sensible cut-off point (like "500 MB" or "1 week"). Report label and case number also are very sensible.

OTOH options like "allow system changes", "verify packages" or "skip post-processing" sound obscure enough to me as an "outsider" that they should have reasonable defaults. "threads" is also over-configuration -- `nproc` is a fine default there for an interactive UI where you are waiting for the result.

So maybe let's play this "you are the expert" vs. "we are the trial subjects and give design feedback" for a few more rounds, until we are both happy?

Thank you!

Reply:
> FWIW, if you think it's useful to discuss this with high bandwidth, let's schedule a call.

That'd probably be a great step. I'll shoot you an email to get something arranged.